### ISSUE TITLE
eBPF causes a coredump

### ISSUE THREAD

**Issue by johnpiers** [netdata/netdata#9957]
removed '/var/lib/systemd/coredump/core.tracker-extract.1000.4fdfe05aeff14eb39d4c0c6a0e239983.238320.1600420595000000.zst'
removed '/var/lib/systemd/coredump/core.tracker-extract.1000.4fdfe05aeff14eb39d4c0c6a0e239983.238346.1600420595000000.zst'
removed '/var/lib/systemd/coredump/core.tracker-extract.1000.4fdfe05aeff14eb39d4c0c6a0e239983.238381.1600420596000000.zst'
removed '/var/lib/systemd/coredump/core.tracker-extract.1000.4fdfe05aeff14eb39d4c0c6a0e239983.238414.1600420597000000.zst'
removed '/var/lib/systemd/coredump/core.tracker-extract.1000.4fdfe05aeff14eb39d4c0c6a0e239983.238438.1600420598000000.zst'
removed '/var/lib/systemd/coredump/core.tracker-extract.1000.4fdfe05aeff14eb39d4c0c6a0e239983.238977.1600420720000000.zst'
removed '/var/lib/systemd/coredump/core.tracker-extract.1000.4fdfe05aeff14eb39d4c0c6a0e239983.238999.1600420720000000.zst'
removed '/var/lib/systemd/coredump/core.tracker-extract.1000.4fdfe05aeff14eb39d4c0c6a0e239983.239030.1600420721000000.zst'
removed '/var/lib/systemd/coredump/core.tracker-extract.1000.4fdfe05aeff14eb39d4c0c6a0e239983.239063.1600420722000000.zst'
removed '/var/lib/systemd/coredump/core.tracker-extract.1000.4fdfe05aeff14eb39d4c0c6a0e239983.239093.1600420723000000.zst'
removed '/var/lib/systemd/coredump/core.tracker-extract.1000.4fdfe05aeff14eb39d4c0c6a0e239983.239571.1600420846000000.zst'
removed '/var/lib/systemd/coredump/core.tracker-extract.1000.4fdfe05aeff14eb39d4c0c6a0e239983.239599.1600420847000000.zst'
removed '/var/lib/systemd/coredump/core.tracker-extract.1000.4fdfe05aeff14eb39d4c0c6a0e239983.239630.1600420847000000.zst'
removed '/var/lib/systemd/coredump/core.tracker-extract.1000.4fdfe05aeff14eb39d4c0c6a0e239983.239660.1600420848000000.zst'
removed '/var/lib/systemd/coredump/core.tracker-extract.1000.4fdfe05aeff14eb39d4c0c6a0e239983.239691.1600420849000000.zst'

**Comment by ilyam8** [comment]
Hi @johnpiers please add `Question summary`, `OS / Environment`, coredump logs, etc

**Comment by johnpiers** [comment]
@ilyam8 Sorry, I'm a bit of a clot. Hopefully I've given you what you need. Thank you.

**Comment by thiagoftsm** [comment]
Hi @johnpiers ,

Please, can you run the following command and give us the output?

```
uname -a; grep -Hv "^#" /etc/*release  # Linux
```

We need to know your kernel version and OS to try to recreate the problem.

Best regards!

**Comment by thiagoftsm** [comment]
@johnpiers ,

I am running more than 2 hours Netdata with `ebpf.plugin` enabled on a Slackware Linux current running kernel `5.8.9` and I cannot recreate the problem.

How did you install your Netdata on your machine? Is the update running automatically?

Best regards

**Comment by johnpiers** [comment]
@thiagoftsm I've always used, bash <(curl -Ss https://my-netdata.io/kickstart.sh) and the auto update has never worked. I've just waited for the nightly update and then just run the script. I meticulously uninstalled Netdata this evening and then did a clean install using the same kickstarter script. I've looked at everything that I can think of but my level has reached it's limit and I don't really know what I'm suppose to look for or what exactly it is I'm looking for or how to repair it. 
TIME                            PID   UID   GID SIG COREFILE  EXE
Sun 2020-09-06 04:17:53 SAST  16518   134   134   6 missing   /usr/libexec/netdata/plugins.d/ebpf.plugin
Mon 2020-09-07 08:22:30 SAST    380   134   134  11 missing   /usr/bin/netdata
Tue 2020-09-08 10:55:19 SAST 229398   134   134   6 missing   /usr/libexec/netdata/plugins.d/ebpf.plugin
Wed 2020-09-09 03:59:26 SAST 229457   134   134   6 missing   /usr/bin/netdata
Thu 2020-09-10 08:07:14 SAST 799576   134   134   6 missing   /usr/libexec/netdata/plugins.d/ebpf.plugin
Fri 2020-09-11 05:12:31 SAST  93365   134   134   6 missing   /usr/libexec/netdata/plugins.d/ebpf.plugin
Sat 2020-09-12 03:38:02 SAST 417485   134   134   6 missing   /usr/libexec/netdata/plugins.d/ebpf.plugin
Sat 2020-09-12 20:30:48 SAST 635540 65534 65534  11 missing   /usr/bin/ntop
Sun 2020-09-13 00:35:14 SAST 688841  1000   985  11 missing   /usr/lib/office6/wps
Sun 2020-09-13 12:12:39 SAST 854728   134   134   6 missing   /usr/libexec/netdata/plugins.d/ebpf.plugin
Mon 2020-09-14 08:20:50 SAST    382   134   134  11 missing   /usr/bin/netdata
Wed 2020-09-16 00:28:11 SAST  26843  1000   985  31 missing   /usr/lib/tracker-extract
Wed 2020-09-16 00:28:11 SAST  26864  1000   985  31 missing   /usr/lib/tracker-extract
Wed 2020-09-16 00:28:12 SAST  26902  1000   985  31 missing   /usr/lib/tracker-extract
Wed 2020-09-16 00:28:46 SAST  27090  1000   985  31 missing   /usr/lib/tracker-extract
Wed 2020-09-16 00:28:46 SAST  27118  1000   985  31 missing   /usr/lib/tracker-extract
Wed 2020-09-16 00:28:47 SAST  27142  1000   985  31 missing   /usr/lib/tracker-extract
Wed 2020-09-16 00:30:13 SAST  27638  1000   985  31 missing   /usr/lib/tracker-extract
Wed 2020-09-16 00:30:13 SAST  27659  1000   985  31 missing   /usr/lib/tracker-extract
Wed 2020-09-16 00:30:14 SAST  27690  1000   985  31 missing   /usr/lib/tracker-extract
Wed 2020-09-16 20:52:21 SAST 300759   134   134   6 missing   /usr/libexec/netdata/plugins.d/ebpf.plugin
Wed 2020-09-16 21:34:10 SAST 318735  1000   985  31 missing   /usr/lib/tracker-extract
Wed 2020-09-16 21:34:11 SAST 318756  1000   985  31 missing   /usr/lib/tracker-extract
Wed 2020-09-16 21:34:12 SAST 318791  1000   985  31 missing   /usr/lib/tracker-extract
Thu 2020-09-17 03:50:39 SAST 411705   134   134   6 missing   /usr/libexec/netdata/plugins.d/ebpf.plugin
Thu 2020-09-17 22:38:45 SAST   1295  1000   985  31 missing   /usr/lib/tracker-extract
Thu 2020-09-17 22:38:46 SAST   1536  1000   985  31 missing   /usr/lib/tracker-extract
Thu 2020-09-17 22:38:47 SAST   1561  1000   985  31 missing   /usr/lib/tracker-extract
Thu 2020-09-17 22:38:48 SAST   1585  1000   985  31 missing   /usr/lib/tracker-extract
Thu 2020-09-17 22:38:49 SAST   1614  1000   985  31 missing   /usr/lib/tracker-extract
Thu 2020-09-17 22:38:49 SAST   1646  1000   985  31 missing   /usr/lib/tracker-extract
Thu 2020-09-17 22:40:52 SAST   3685  1000   985  31 missing   /usr/lib/tracker-extract
Thu 2020-09-17 22:40:53 SAST   3734  1000   985  31 missing   /usr/lib/tracker-extract
Thu 2020-09-17 22:40:54 SAST   3760  1000   985  31 missing   /usr/lib/tracker-extract
Thu 2020-09-17 22:40:55 SAST   3802  1000   985  31 missing   /usr/lib/tracker-extract
Thu 2020-09-17 22:40:55 SAST   3839  1000   985  31 missing   /usr/lib/tracker-extract
Thu 2020-09-17 22:42:58 SAST   4826  1000   985  31 missing   /usr/lib/tracker-extract
Thu 2020-09-17 22:42:59 SAST   4853  1000   985  31 missing   /usr/lib/tracker-extract
Thu 2020-09-17 22:43:00 SAST   4890  1000   985  31 missing   /usr/lib/tracker-extract
Thu 2020-09-17 22:43:00 SAST   4917  1000   985  31 missing   /usr/lib/tracker-extract
Thu 2020-09-17 22:43:01 SAST   4945  1000   985  31 missing   /usr/lib/tracker-extract
Thu 2020-09-17 22:45:03 SAST   5534  1000   985  31 missing   /usr/lib/tracker-extract
Thu 2020-09-17 22:45:04 SAST   5557  1000   985  31 missing   /usr/lib/tracker-extract
Thu 2020-09-17 22:45:05 SAST   5593  1000   985  31 missing   /usr/lib/tracker-extract
Thu 2020-09-17 22:45:05 SAST   5624  1000   985  31 missing   /usr/lib/tracker-extract
Thu 2020-09-17 22:45:06 SAST   5653  1000   985  31 missing   /usr/lib/tracker-extract
Thu 2020-09-17 22:47:08 SAST   6166  1000   985  31 missing   /usr/lib/tracker-extract
Thu 2020-09-17 22:47:09 SAST   6197  1000   985  31 missing   /usr/lib/tracker-extract
Thu 2020-09-17 22:47:10 SAST   6230  1000   985  31 missing   /usr/lib/tracker-extract
Thu 2020-09-17 22:47:10 SAST   6252  1000   985  31 missing   /usr/lib/tracker-extract
 
The above is when it first started from "coredumpctl list" to the present: A total of about 3500 coredumps. The ones listed as missing are deleted (cleared)
Fri 2020-09-18 21:53:16 SAST  51432  1000   985  31 present   /usr/lib/tracker-extract
Fri 2020-09-18 21:53:17 SAST  51452  1000   985  31 present   /usr/lib/tracker-extract
Fri 2020-09-18 21:53:18 SAST  51486  1000   985  31 present   /usr/lib/tracker-extract
Fri 2020-09-18 21:53:19 SAST  51516  1000   985  31 present   /usr/lib/tracker-extract
Fri 2020-09-18 21:53:20 SAST  51544  1000   985  31 present   /usr/lib/tracker-extract
Fri 2020-09-18 21:55:22 SAST  51751  1000   985  31 present   /usr/lib/tracker-extract
Fri 2020-09-18 21:55:23 SAST  51771  1000   985  31 present   /usr/lib/tracker-extract
Fri 2020-09-18 21:55:24 SAST  51801  1000   985  31 present   /usr/lib/tracker-extract
Fri 2020-09-18 21:55:25 SAST  51832  1000   985  31 present   /usr/lib/tracker-extract
Fri 2020-09-18 21:55:26 SAST  51861  1000   985  31 present   /usr/lib/tracker-extract
Fri 2020-09-18 21:57:28 SAST  52053  1000   985  31 present   /usr/lib/tracker-extract
Fri 2020-09-18 21:57:29 SAST  52088  1000   985  31 present   /usr/lib/tracker-extract
Fri 2020-09-18 21:57:29 SAST  52109  1000   985  31 present   /usr/lib/tracker-extract
Fri 2020-09-18 21:57:30 SAST  52140  1000   985  31 present   /usr/lib/tracker-extract
Fri 2020-09-18 21:57:31 SAST  52165  1000   985  31 present   /usr/lib/tracker-extract
Fri 2020-09-18 21:59:34 SAST  52473  1000   985  31 present   /usr/lib/tracker-extract
Fri 2020-09-18 21:59:35 SAST  52496  1000   985  31 present   /usr/lib/tracker-extract
Fri 2020-09-18 21:59:36 SAST  52532  1000   985  31 present   /usr/lib/tracker-extract
Fri 2020-09-18 21:59:37 SAST  52563  1000   985  31 present   /usr/lib/tracker-extract
Fri 2020-09-18 21:59:38 SAST  52587  1000   985  31 present   /usr/lib/tracker-extract

**Comment by thiagoftsm** [comment]
I will request more one thing for you, please, can you zip a core dump for ebpf.plugin and attach to the next answer?

@Ferroin you also work with Arch Linux, can you give us a tip about this error? I am not sure if this can have relationship with systemd.

Best regards!

**Comment by johnpiers** [comment]
Sorry I'm only getting back to you guys now, I've been on a mission trying to see if I could solve this myself and then the internet went down. 

The thing "causing" the dump is (tracker) tracker-miner-fs, that's found in usr/lib/tracker-miner-fs. As soon as I pkill tracker-miner-fs, whatever is causing the memory dump "obviously" stops.

Then I went about disabling everything using the method below,found on GitHub Gist, this disables everything but the tracker-store, which even though it has a .desktop file, seems tenacious and starts up anyway. However, nothing gets indexed.

tracker daemon -t
cd ~/.config/autostart
cp -v /etc/xdg/autostart/tracker-*.desktop ./
for FILE in tracker-*.desktop; do echo Hidden=true >> $FILE; done
rm -rf ~/.cache/tracker ~/.local/share/tracker

After this is done, tracker-store will still start on the next boot. However, nothing will be indexed.

If you run 'tracker status' you should get this:

$ tracker status
Currently indexed: 0 files, 0 folders
Remaining space on database partition: 123 GB (78.9%)
All data miners are idle, indexing complete

Although this is probably not such a good idea and more of a workaround, if it doesn't cause any grief, and also far less invasive, then so be it. If one of you guys can offer any more advice in terms of what I've done, or if there is another way, then that would be awesome.

Thanks very much for the assistance.
Best regards,
Piers.

**Comment by ilyam8** [comment]
> eBPF causes a coredump

> tracker-miner-fs

> metadata database, indexer and search tool - filesystem indexer


how is that connected with netdata/ebpf?

**Comment by johnpiers** [comment]
Apologies, as I said initially, I had no idea why I was getting these memory dumps:
Sun 2020-09-06 04:17:53 SAST 16518 134 134 6 missing /usr/libexec/netdata/plugins.d/ebpf.plugin
Mon 2020-09-07 08:22:30 SAST 380 134 134 11 missing /usr/bin/netdata
Tue 2020-09-08 10:55:19 SAST 229398 134 134 6 missing /usr/libexec/netdata/plugins.d/ebpf.plugin
Wed 2020-09-09 03:59:26 SAST 229457 134 134 6 missing /usr/bin/netdata
Thu 2020-09-10 08:07:14 SAST 799576 134 134 6 missing /usr/libexec/netdata/plugins.d/ebpf.plugin
Fri 2020-09-11 05:12:31 SAST 93365 134 134 6 missing /usr/libexec/netdata/plugins.d/ebpf.plugin
Sat 2020-09-12 03:38:02 SAST 417485 134 134 6 missing /usr/libexec/netdata/plugins.d/ebpf.plugin
Sat 2020-09-12 20:30:48 SAST 635540 65534 65534 11 missing /usr/bin/ntop
Sun 2020-09-13 00:35:14 SAST 688841 1000 985 11 missing /usr/lib/office6/wps
Sun 2020-09-13 12:12:39 SAST 854728 134 134 6 missing /usr/libexec/netdata/plugins.d/ebpf.plugin
Mon 2020-09-14 08:20:50 SAST 382 134 134 11 missing /usr/bin/netdata

And I've implemented a way of stopping the memory dump but obviously not the cause of it. This only started core dumping as per the above entries and continued up until I disabled "tracker" after posting the question. As I also said I was using a workaround to get rid of the symptom but not the cause. I don't know how to fix it, in fact I have no idea.  Sorry about that.
